stock/src/main/java/nl/tudelft/wdm/group1/stock/web/StockController.java

```diff
@@ -39,4 +36,29 @@ public StockItem addStockItem() {
     public StockItem getStockItem(@PathVariable(value = "id") UUID id) throws ResourceNotFoundException {
         return stockItemRepository.find(id);
     }
+
+    @PostMapping("/{id}/subtract/{amount}")
+    public StockItem substractStockItemAmount(
+            @PathVariable(value = "id") UUID id,
+            @PathVariable(value = "amount") int amount
+    ) throws ResourceNotFoundException, InsufficientStockException {
+        StockItem item = stockItemRepository.find(id);
+        item.subtractStock(amount);
+        producer.send(item);
```

Comment:
Subtraction/addition need different events, see https://github.com/casperboone/wdm-kafka-microservices/pull/2

Reply:
Got it.